### PR TITLE
Happy thoughts app Vittoria

### PR DESCRIPTION
**Pull request by vittoriamatteoli**
## Netlify link
[happy-vibes-app-vicky.netlify.app](https://happy-vibes-app-vicky.netlify.app/)

## Collaborators
[vittoriamatteoli]

### CODE REVIEW

**Review by dzc1 — APPROVED** [review]
Vittoria first and foremost, i gotta say its been extremely cool reviewing your project. 

`ThoughtsApp.jsx` being the 'top-level' file with all the [power](https://giphy.com/gifs/jim-carrey-power-bruce-almighty-A9grgCQ0Dm012) of your app is neatly distributed. Loved the approach on breaking down each logic that is loaddedd through the useEffect individually, [mad respect](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExem05djUzZm5kemIybTQ3NTR2dnE5MGU5emw4bGM5NTA4bXdrYnEwcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/eJF2H9ZiPTZBEWC3Zv/giphy.gif). 

Lastly, local-storage dependency is amazing, to put simply. V, this will help you so much when it comes time to implement backend structures. 

I do haave to say, goodd workaround on line 73 of `ThoughtsApp.jsx` , pure js. if you want comment that line and add the variable `thoughts` within the array of line 33 much like this: 

```
// inside the useEffect hook I fetch the API, I sort the data in descending order and I update the list with the sorted item/set loading during the fetching.
  useEffect(() => {
    fetch(thoughtsAPI)
      .then((response) => response.json())
      .then((list) => {
        const sortedList = list.sort(
          (a, b) => new Date(b.createdAt) - new Date(a.createdAt)
        );
        setList(sortedList);
        setLoading(false);
      })
      .catch((error) => {
        setError(error);
        setLoading(false);
      });
  }, [thought]);
```

Although the same output is provided, this uses the useEffect to it's pure form and a bit less code. Which can seem odd at the beginning but hey thats why we have React. 

Nevertheless, amazing work , hoping that you are enjoying using the react framework!

**Comment by vittoriamatteoli** [comment]
Thank you Diego! I didn't realize that I could add the thought as a dependencies, I refactored the code many times to implement new functionalities following the advanced goals  and I clearly missed that , thanks !!

File: src/Components/Form/Form.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        <textarea
+          rows="4"
+          cols="50"
+          value={thought}
+          onChange={onNewThought}
+          placeholder="Write some text..."
+        />
+      </label>
+      <span
+        className="wordCounter"
```
Comment (sofia32057) [inline src/Components/Form/Form.jsx:16]: Hey, as a heads up - even if we're doing jsx I believe the Good Code Guidelines still wants us to write class names with kebab-case (word-counter). This applies throughout your repo :)

File: src/Components/List/List.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export const List = ({ list, formatDate, handleNewLike }) => {
+  return (
+    <div className="listWrapper">
+      {list.map((thought) => (
+        <div className="thoughtContainer" key={thought._id}>
+          <h2 className="message fadeIn">{thought.message}</h2>
+          <div className="dateWrapper">
+            <div className="likesWrapper">
+              <button
+                className={thought.hearts > 0 ? 'likeBtn' : 'noLikesBtn'}
```
Comment (sofia32057) [inline src/Components/List/List.jsx:11]: Smart way to set the button color!

File: src/Components/ThoughtsApp/ThoughtsApp.jsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    const storedLikes = localStorage.getItem('likeCount')
+    if (storedLikes) {
+      setLikeCount(parseInt(storedLikes, 10))
+    }
+  }, [])
+  //update localStorage when a new like it's added
+  // it adds to local storage a pair of key-value , in this case I create in the storage an object with a key=likecount value=likecount.toString() and does that whenever the likecount state change(see setLikeCount).
+  useEffect(() => {
+    localStorage.setItem('likeCount', likeCount.toString())
+  }, [likeCount])
```
Comment (sofia32057) [inline src/Components/ThoughtsApp/ThoughtsApp.jsx:55]: So awesome of you figuring out local storage!

**Review by sofia32057 — COMMENTED** [review]
Hi Vittoria,
I'm so sorry for the late review!

You did awesome as always :) I love how you conquered local storage. 🔥

There are some naming conventions you should maybe take a look at (but since Diego didn't mention it I guess it's ok). Also, I think you should push yourself and create more components, just to level up a bit. 

Happy easter! 🐣

**Pull request closed without merging**